app/parsers/bulkrax/parser_export_record_set.rb

```diff
@@ -134,30 +134,34 @@ def collections
                          end
       end
 
+      SOLR_QUERY_PAGE_SIZE = 512
+
       # @note In most cases, when we don't have any candidate file sets, there is no need to query SOLR.
       #
       # @see Bulkrax::ParserExportRecordSet::Importer#file_sets
-      def file_sets
-        @file_sets ||= if candidate_file_set_ids.empty?
-                         []
-                       else
-                         ActiveFedora::SolrService.query(file_sets_query, **file_sets_query_kwargs)
-                       end
-      end
-
+      #
       # Why can't we just use the candidate_file_set_ids?  Because Hyrax is pushing child works into the
       # `file_set_ids_ssim` field.
       #
       # For v2.9.x of Hryax; perhaps this is resolved.
       #
       # @see https://github.com/scientist-softserv/britishlibrary/issues/289
       # @see https://github.com/samvera/hyrax/blob/64c0bbf0dc0d3e1b49f040b50ea70d177cc9d8f6/app/indexers/hyrax/work_indexer.rb#L15-L18
-      def file_sets_query
-        %(has_model_ssim:#{Bulkrax.file_model_class} AND id:("#{candidate_file_set_ids.join('" OR "')}") #{extra_filters})
-      end
-
-      def file_sets_query_kwargs
-        { fl: "id", method: :post, rows: candidate_file_set_ids.size }
+      def file_sets
+        @file_sets ||= if candidate_file_set_ids.empty?
+                         []
+                       else
+                         results = []
+                         candidate_file_set_ids.each_slice(SOLR_QUERY_PAGE_SIZE) do |ids|
+                           fsq = "has_model_ssim:#{Bulkrax.file_model_class} AND id:(\"" + ids.join('" OR "') + "\")"
+                           fsq += extra_filters if extra_filters.present?
+                           results += ActiveFedora::SolrService.query(
+                             fsq,
+                             { fl: "id", method: :post, rows: ids.size }
+                           )
+                         end
+                         results
+                       end
```

Comment:
this makes sense to me based on the PR description. 

the spacing/indentation of lines 152-164 looks off for me.. not sure if its just github being weird though

Reply:
The indentation is atypical due to paranthesis placement; however all of this appeases Rubocop.